src/python/gudhi/simplex_tree.pyx

```diff
@@ -285,6 +285,24 @@ cdef class SimplexTree:
             ct.append((v, filtered_simplex.second))
         return ct
 
+    def get_boundaries(self, simplex):
+        """This function returns a generator with the boundaries of a given N-simplex.
+
+        :param simplex: The N-simplex, represented by a list of vertex.
+        :type simplex: list of int.
+        :returns:  The (simplices of the) boundary of a simplex
+        :rtype:  generator with tuples(simplex, filtration)
+        """
+        cdef pair[Simplex_tree_boundary_iterator, Simplex_tree_boundary_iterator] it =  self.get_ptr().get_boundary_iterators(simplex)
+
+        try:
+            while it.first != it.second:
+                yield self.get_ptr().get_simplex_and_filtration(dereference(it.first))
+                preincrement(it.first)
+        except RuntimeError:
+            print("simplex not found - cannot find boundaries")
```

Comment:
I would remove the try-except-print. Letting the exception propagate seems better.

Reply:
Let the exception propagate on 41869710